scripts/sanity_chk/harness.py

```diff
@@ -121,6 +129,48 @@ def handle(self, line):
 
         self.process_test(line)
 
+class Pytest(Harness):
+    def handle(self, line):
+        cmd = [
+			'pytest',
+			'-s',
+			self.source_dir + '/' + self.pytest_root,
+			'--cmdopt',
+			self.running_dir,
+			'--junit-xml',
+			self.running_dir + '/report.html'
+        ]
+
+        with subprocess.Popen(cmd) as proc:
+            try:
+                proc.wait(15)
+                tree = ET.parse(self.running_dir + "/report.html")
+                root = tree.getroot()
+                for child in root:
+                    if child.tag == 'testsuite':
+                        if child.attrib['failures'] != '0':
+                            self.state = "failed"
+                        elif child.attrib['skipped'] != '0':
+                            self.state = "skipped"
+                        else:
```

Comment:
there also can be `errors` != 0 in the junit report, which you are not capturing now. Please add the case for it.

Reply:
added in #33227